link/uprobe_test.go

```diff
@@ -88,6 +88,20 @@ func TestUprobeExtWithOpts(t *testing.T) {
 	defer up.Close()
 }
 
+func TestUprobeWithPid(t *testing.T) {
+	prog, err := ebpf.NewProgram(&kprobeSpec)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer prog.Close()
+
+	up, err := bashEx.Uprobe(bashSym, prog, &UprobeOptions{Pid: -1})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer up.Close()
+}
+
```

Comment:
I'm missing a negative test here, e.g. running 2 bash processes and only tracing one of them to make sure the PID filter behaves as expected. I've been looking for a way to be able to clone() -> get the PID -> hook up the uprobe -> exec(), but I couldn't find a way to do this in Go out of the box. We could do a `bash -c 'sleep 1; exec bash` to give the test suite time to call `cmd.Start()`, obtain the PID and open the uprobe, but we want to avoid sleeping if at all possible. And it might still race.

Could we maybe trace pid1 (or another equally unlikely PID) and check for absence of a uprobe execution instead? That way, at least we're sure that the parameter is correctly propagated and that filtering works.

Reply:
I've made an integration test similar to the ones we already have, let me know what you think

Reply:
Solid, thanks! :+1: